tensorflow/compiler/xla/service/pattern_matcher.h

```diff
@@ -1268,36 +1268,36 @@ class HloInstructionPatternOpcodeImpl {
 
 // An HloInstructionPattern implementation that optionally matches a unary
 // operand with a given opcode before matching a given pattern.
-template <typename OperandType, typename OperandImpl>
+template <typename PatternType, typename PatternImpl>
 class HloInstructionPatternOptionalUnaryOpImpl {
  public:
   explicit HloInstructionPatternOptionalUnaryOpImpl(
       absl::Span<const HloOpcode> opcodes,
-      const HloInstructionPattern<OperandType, OperandImpl>& operand)
-      : opcodes_(opcodes), operand_(operand) {}
+      const HloInstructionPattern<PatternType, PatternImpl>& pattern)
+      : opcodes_(opcodes), pattern_(pattern) {}
 
   bool Match(::xla::HloInstruction* inst, MatchOption option) const {
     // Compare the opcode of the instruction with the entries of opcodes_.
     if (c_linear_search(opcodes_, inst->opcode())) {
       // Additionally, the operand of the instruction must match the given
       // operand pattern.
-      if (operand_.Match(HloOperand(inst, 0), option)) {
+      if (pattern_.Match(HloOperand(inst, 0), option)) {
         return true;
       } else {
-        EXPLAIN << " \nExpected the operand of the instruction to match the "
-                   "prescribed operand pattern. ";
+        EXPLAIN << " \nThe operand of the HloInstruction doesn't match "
+                << pattern_.ToString() << " and the ";
```

Comment:
This will print pattern_ twice -- once here, and once at the end of the function.

I don't think you want that.  In the testcases you have, the pattern_ is simple and it's ok.  But the pattern can be arbitrarily complicated (many lines long).

Reply:
I think so -- it's better to print the pattern string only once.  We should be able to come up with some phrasing for the `EXPLAIN`s so that it makes sense when you read it.